File: tests/fixtures/go/shadowing.go
package sample

func target() {}

func Shadowing(factory func() func()) {
	target := func() {
		target()
	}
	target()

	target = factory()
	target()

	target, extra := factory(), factory()
	target()
	extra()

	var first, second = factory(), factory()
	first()
	second()

	{
		nested := factory()
		nested()
	}
	nested()

	object.field = factory()
	object()
	values[0] = factory()
	values()
}

File: tests/fixtures/rust/shadowing.rs
fn target() {}

pub fn shadowing(factory: impl Fn() -> fn()) {
    let target = || {
        target();
    };
    target();

    let mut callback = || {};
    callback();
    callback = factory();
    callback();

    let (first, second) = (factory(), factory());
    first();
    second();

    {
        let nested = factory();
        nested();
    }
    nested();

    object.callback = factory();
    object();
    values[0] = factory();
    values();
}
